src/aks-preview/azcli-aks-live-test/az_aks_tool/ext.py

```diff
@@ -0,0 +1,88 @@
+
+import utils
+
+
+def get_ext_test_index(mod_name):
+    # path table & name index
+    path_table = get_path_table()
+    command_modules = path_table['mod']
+    extensions = path_table["ext"]
+    inverse_name_table = get_name_index(invert=True)
+
+    # import_name & mod_data
+    aks_preview_mod_path = extensions[mod_name]
+    glob_pattern = os.path.normcase(
+        os.path.join("{}*".format(EXTENSION_PREFIX)))
+    file_path = glob.glob(os.path.join(aks_preview_mod_path, glob_pattern))[0]
+    import_name = os.path.basename(file_path)
+    mod_data = {
+        "alt_name": inverse_name_table[mod_name],
+        "filepath": os.path.join(file_path, "tests", "latest"),
+        "base_path": "{}.tests.{}".format(import_name, "latest"),
+        "files": {}
+    }
+
+    # azdev hook
+    ext_test = _discover_module_tests(import_name, mod_data)
+    ext_test_index = ext_test["files"]
+    return ext_test_index
+
+
+def get_ext_test_cases(ext_test_index, ext_matrix, ext_extra_coverage):
+    ext_test_cases = []
+    ext_coverage = ext_matrix["coverage"]
+    for fileName, className in ext_coverage.items():
+        for c in className:
+            ext_test_cases.extend(ext_test_index[fileName][c])
+    # extra coverage
+    if ext_extra_coverage:
+        # method 1: fileName.className
+        file_class_pairs = utils.extract_file_class_pairs(ext_extra_coverage)
+        valid_file_class_pairs = utils.filter_valid_file_class_pairs(
+            file_class_pairs, ext_test_index)
+        for valid_pair in valid_file_class_pairs:
+            ext_test_cases.extend(
+                ext_test_index[valid_pair[0]][valid_pair[1]])
+        # method 2: test cases
+        ext_test_cases.extend(utils.filter_valid_test_cases(
+            ext_extra_coverage, ext_test_index))
+    return list(set(ext_test_cases))
+
+
+def get_ext_exclude_test_cases(ext_test_index, ext_matrix, ext_filter):
+    ext_exclude_test_cases = []
+    ext_exclude = ext_matrix["exclude"]
+    if not ext_filter:
+        for k, v in ext_exclude.items():
+            ext_exclude_test_cases.extend(v)
+    else:
+        # method 1: matrix exclude key
+        for k, v in ext_exclude.items():
+            if k in ext_filter or "default" in ext_filter:
+                ext_exclude_test_cases.extend(v)
+        # method 2: fileName.className
+        file_class_pairs = utils.extract_file_class_pairs(ext_filter)
+        valid_file_class_pairs = utils.filter_valid_file_class_pairs(
+            file_class_pairs, ext_test_index)
+        for valid_pair in valid_file_class_pairs:
+            ext_exclude_test_cases.extend(
+                ext_test_index[valid_pair[0]][valid_pair[1]])
+        # method 3: test cases
+        ext_exclude_test_cases.extend(
+            utils.filter_valid_test_cases(ext_filter, ext_test_index))
+    return list(set(ext_exclude_test_cases))
+
+
+def get_ext_filted_test_cases(ext_test_cases, ext_exclude_test_cases):
```

Comment:
unit tests for these?

Reply:
updated in commit [f2fe148](https://github.com/Azure/azure-cli-extensions/pull/3292/commits/f2fe1484b1f5ecdafccdb2151169d226899d24aa).

Reply:
> updated in commit [f2fe148](https://github.com/Azure/azure-cli-extensions/pull/3292/commits/f2fe1484b1f5ecdafccdb2151169d226899d24aa).

could you please also setup one unit test pipeline to run your tests? and we can also add real unit tests for the prod codes in that pipeline in the future.

Reply:
sure, the unit test pipeline is [here](https://dev.azure.com/msazure/CloudNativeCompute/_build?definitionId=212357&_a=summary).